tracer/test/Datadog.Trace.SourceGenerators.Tests/AspectsDefinitionsGeneratorTests.cs

```diff
@@ -484,6 +494,65 @@ internal static partial class AspectDefinitions
             Assert.Empty(diagnostics);
         }
 
+        [Fact]
+        public void CanGenerateNativeAspectsDefinitionForSimpleClass()
+        {
+            const string input = """
+using Datadog.Trace.SourceGenerators;
+using Datadog.Trace.Iast;
+using Datadog.Trace.Iast.Dataflow;
+
+namespace MyTests;
+
+[Datadog.Trace.Telemetry.TargetFrameworkMoniker("TestTFM")]
+[AspectClass("mscorlib,netstandard,System.Private.CoreLib", new[] { AspectFilter.StringOptimization })]
+public class TestAspectClass
+{ 
+    [AspectMethodReplace("System.String::Concat(System.String,System.String)", AspectFilter.StringLiterals_Any)]
+    public static string Concat(string target, string param1)
+    {
+        return string.Concat(target, param1);
+    }
+    [AspectMethodReplace("System.String::Concat(System.Object,System.Object,System.Object)", AspectFilter.StringLiterals_Any)]
+    public static string Concat(object target, object param1, object param2)
+    {
+        return string.Concat(target, param1, param2);
+    }
+}
+""";
+
+            const string expected = Constants.FileHeaderCpp + """"
+#pragma once
+#include "generated_definitions.h"
+
+namespace trace
+{
+
+std::vector<WCHAR*> g_callSites_TestTFM=
+{
+(WCHAR*)WStr("[AspectClass(\"mscorlib,netstandard,System.Private.CoreLib\",[StringOptimization],Propagation,[])] MyTests.TestAspectClass"),
+(WCHAR*)WStr("  [AspectMethodReplace(\"System.String::Concat(System.String,System.String)\",\"\",[0],[False],[StringLiterals_Any],Default,[])] Concat(System.String,System.String)"),
+(WCHAR*)WStr("  [AspectMethodReplace(\"System.String::Concat(System.Object,System.Object,System.Object)\",\"\",[0],[False],[StringLiterals_Any],Default,[])] Concat(System.Object,System.Object,System.Object)"),
+};
+
+std::vector<WCHAR*> g_callSites_TestTFM_Rasp=
```

Comment:
I would add a test with some RASP aspects

Reply:
Added specific rasp test in de52372590a36c1b9d2dbe83b9e8bfbfbff2b290